File: js/Tester.js
function log(s) { 
    console.log(`|${s}|`)
}

function verdict2(actual, expected) {
    const a = JSON.stringify( actual )
    const b = JSON.stringify(expected)
    let isFine = "FAIL"
    if ( a == b ) { 
        isFine = "PASS"
    } 
    console.log(`${isFine} |${actual}|  ----> |${expected}|`)
}


function verdict(actual, expected) {
    let isFine = "FAIL"
    if ( actual == expected ) { 
        isFine = "PASS"
    } 
    console.log(`${isFine} |${actual}|  ----> |${expected}|`)
}

const divmod = (numerator, denominator) => {
    /*
    Floor Division
    quotient = n // d 
    remainder = n % d
    */
    return {
        quotient:Math.floor(numerator/denominator),
        remainder:numerator % denominator
    }
}
const isAlphaNumeric = ( word ) => { 
    if (word.match(/^[0-9a-zA-Z]+$/)) {
        return true
    } else {
        return false
    }
}
module.exports = {log, verdict, isAlphaNumeric, divmod, verdict2}


//https://github.com/kermitt/challenges/blob/main/py/FlipWordsButNotSentence.py
//https://github.com/kermitt/challenges/blob/main/java/FlipWordsButNotSentence.java

File: java/Tester.java
public class Tester {
	static void log(String s) { 
		System.out.println(s);
	}
	static void log(String[] ary) { 
		String a = "["; 
		for ( String aa : ary ) {
			a += aa + " ";
		}
		a = a.trim();
		a += "]";
		System.out.println(a);

	}

	static void verdict(String[] actual, String[] expected) {
		String didPass = "FAIL";
		String a = "["; 
		for ( String aa : actual ) {
			a += aa + " ";
		}
		a = a.trim();
		a += "]";
		String e = "["; 
		for ( String ee : expected ) {
			e += ee + " "; 
		}
		e = e.trim(); 
		e += "]";

		if ( a.equals(e) ) {
			didPass = "PASS";
	   }


		String s = String.format("%s %s  !!--> %s", didPass, a, e);
		System.out.println(s);
	}

	static void verdict(String actual, String expected) {
		String didPass = "FAIL";
		if ( actual.equals(expected)) {
			 didPass = "PASS";
		}
		String s = String.format("%s |%s|   ---> |%s|", didPass, actual, expected);
		System.out.println(s);
	}
//	public static void main(String...strings) {
//		Tester t = new Tester();
//		t.verdict("ONE",  "ONE");
//		t.verdict("ONE",  "TWO");
//	}
}
